R/scatter_plot_utils.R

```diff
@@ -398,3 +399,105 @@ simple_model_plot <- function(
 
   return(plot_model)
 }
+
+
+#' Layout attributes updates
+#'
+#'
+#' @param plot_model a plot_ly or ggplot object
+#' @param attribute_name string, name or regex of the name of attribute(s) to
+#' update, extracted from the layout of the plot.
+#' @param attributes list, new attributes to add in the `layoutAttrs` of the
+#' plot (for example, `list(type = "log")` to transform axis to log scale)
+#'
+#' @noRd
+#' @importFrom purrr map
+plot_model_layout_attr <- function(plot_model, attribute_name, attributes) {
+  attr_name <- sort(grep(attribute_name, names(plot_model$x$layout),
```

Comment:
What happens if the `attribute_name` does not match any attributes in `layout`? Or the resulting vector doesn't match the length of `attributes` or the sorting changes how the names map onto the `attributes` elements?

Reply:
It might be good to explain these expectations and restrictions in the comments. Just so if someone else ever had to work with internal source code, the expectations of the function is clear.

Reply:
I updated the documentation:
- add information about the `attribute_name` being extracted from layout
- made is clearer that `attributes` should be a list
- remove `sort` that was just cosmetic.